### PR TITLE
Contract deployment

### PR DESCRIPTION
**Pull request by roman-khimov**
### Problem

#474

### Solution

Merge button.

### CODE REVIEW

File: pkg/rpc/client.go [hunk 1]
```diff
@@ -152,6 +155,29 @@ func (c *Client) SetClient(cli *http.Client) {
...
 }
 
+// CalculateInputs creates input transactions for the specified amount of given
+// asset belonging to specified address. This implementation uses GetUnspents
+// JSON-RPC call internally, so make sure your RPC server suppors that.
+func (c *Client) CalculateInputs(address string, asset util.Uint256, cost util.Fixed8) ([]transaction.Input, util.Fixed8, error) {
+	var utxos core.UnspentBalances
+
+	resp, err := c.GetUnspents(address)
+	if err != nil || resp.Error != nil {
```
Comment (volekerb) [inline pkg/rpc/client.go]: why GetUnspents just can't return err, ie having check for resp.Err inside there? 
and here leave only check for err != nil

File: cli/smartcontract/smart_contract.go [hunk 2]
```diff
@@ -135,7 +136,17 @@ func initSmartContract(ctx *cli.Context) error {
...
+		details.Parameters = make([]rpc.StackParamType, 2)
+		details.Parameters[0] = rpc.String
+		details.Parameters[1] = rpc.Array
+
+		project := &ProjectConfig{Contract: details}
+		b, err := yaml.Marshal(project)
+		if err != nil {
+			return cli.NewExitError(err, 1)
+		}
+		if err := ioutil.WriteFile(filepath.Join(basePath, "neo-go.yml"), b, 0644); err != nil {
```
Comment (volekerb) [inline cli/smartcontract/smart_contract.go]: nit: probably next time move "neo-go.yml" to variable or config

File: pkg/rpc/txBuilder.go [hunk 3]
```diff
@@ -106,3 +110,55 @@ func GetInvocationScript(tx *transaction.Transaction, wif *keys.WIF) ([]byte, er
...
 	return append([]byte{pushbytes64}, signature...), nil
 }
+
+// CreateDeploymentScript returns a script that deploys given smart contract
+// with its metadata.
+func CreateDeploymentScript(avm []byte, contract *ContractDetails) ([]byte, error) {
+	var props smartcontract.PropertyState
+
+	script := new(bytes.Buffer)
+	if err := vm.EmitBytes(script, []byte(contract.Description)); err != nil {
```
Comment (volekerb) [inline pkg/rpc/txBuilder.go]: nit: will vote to move it to some abstraction in the future refactoring.

File: pkg/rpc/client.go (hunk 1 above)
Comment (roman-khimov) [reply pkg/rpc/client.go]: Yeah, but all the other client calls are currently following this convention. It's strange, I think it's wrong, but at this moment we're better doing things consistently and then we'll solve it for all calls.

File: cli/smartcontract/smart_contract.go (hunk 2 above)
Comment (roman-khimov) [reply cli/smartcontract/smart_contract.go]: It's an init command that creates a new directory with some default setup, so it's OK for it to have this file name hardcoded, making it configurable gives nothing to the user.

Comment (volekerb) [reply cli/smartcontract/smart_contract.go]: for user sure, but for developer it can give readability and idea why there is such name.

**Pull request merged**